pkg/engine/defaults.go

```diff
@@ -730,8 +731,46 @@ import (
 	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/zipcodebase"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/zonkafeedback"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/zulipchat"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
 )
 
+// CustomDetectors returns a list of detectors that are enabled by default, but
+// can be overridden by the user.
+func CustomDetectors(ctx context.Context, urls map[string][]string) []detectors.Detector {
+	defaultDetectors := DefaultDetectors()
+	if len(urls) == 0 {
+		return defaultDetectors
+	}
+
+	for i, detector := range defaultDetectors {
+		switch detector.Type() {
+		case detectorspb.DetectorType_Github:
+			githubUrls, ok := urls["github"]
+			if !ok {
+				ctx.Logger().V(2).Info("no GitHub urls to ignore")
+				continue
+			}
+			ctx.Logger().V(2).Info("ignoring GitHub urls: %v", githubUrls)
+			defaultDetectors[i] = github.New(github.WithVerifierURLs(githubUrls, true))
+		case detectorspb.DetectorType_Gitlab:
+			gitlabUrls, ok := urls["gitlab"]
+			if !ok {
+				ctx.Logger().V(2).Info("no GitLab urls to ignore")
+				continue
+			}
+			ctx.Logger().V(2).Info("ignoring GitLab urls: %v", gitlabUrls)
+			defaultDetectors[i] = gitlabv2.New(gitlabv2.WithVerifierURLs(gitlabUrls, true))
+			defaultDetectors[i] = gitlab.New(gitlab.WithVerifierURLs(gitlabUrls, true))
```

Comment:
**Suggestion:** The first line is being overwritten by the second line. We should be able to differentiate between `v1` and `v2` with the `Version()` method. You might have to do a type check to get access to that method:

```go
versioner, ok := detector.(detectors.Versioner)
switch versioner.Version() {
case 1: // ...
case 2: // ...
}
```

Reply:
I just remembered that `github` also has two versions. I think the packages are `github` and `github_old`, so we'll need to do the same version check for the github case.

Reply:
do both versions run when the detectors get run? if so, do we need to check what version it is? because we'd be updating the endpoints for the same DetectorType, and all the versions would pick it up, right?